Fix lint errors/warning

### Goal
No linting or formatting error/warning

### Description
Getting some lint warnings as
```bash
> eluxo-backend@1.0.0 lint
> eslint .


/Users/rajantripathi/Desktop/my-stack/eluxo-backend/app.js
  23:3  warning  Unexpected console statement  no-console
  24:3  warning  Unexpected console statement  no-console
  29:3  warning  Unexpected console statement  no-console
  31:5  warning  Unexpected console statement  no-console

/Users/rajantripathi/Desktop/my-stack/eluxo-backend/server/plugins/crash-handler.js
  3:5  warning  Unexpected console statement  no-console
  9:5  warning  Unexpected console statement  no-console

/Users/rajantripathi/Desktop/my-stack/eluxo-backend/server/plugins/database.js
  18:5  warning  Unexpected console statement  no-console
  22:7  warning  Unexpected console statement  no-console
  27:5  warning  Unexpected console statement  no-console

/Users/rajantripathi/Desktop/my-stack/eluxo-backend/server/plugins/error-handler.js
  3:5  warning  Unexpected console statement  no-console

✖ 10 problems (0 errors, 10 warnings)
```
We have new service names `services/logger`, we have to replace `console` with our `logger` service.

### Instruction
**Steps**
```bash
git fetch origin
git checkout fix/adiigithub/lint-errors
```

**Example**
replace 
```js
console.info(`Server is running at http://${host}:${port}`);
```
with 

```js
const logger = require('./service/logger');


logger.info(`Server is running at http://${host}:${port}`);
```

--- a/.husky/pre-commit
+++ b/.husky/pre-commit
@@ -1,1 +1,1 @@
-npm run lint && npm run format
+npm run precommit

--- a/app.js
+++ b/app.js
@@ -1,5 +1,6 @@
 const http = require('http');
 const { setupServer, applyErrorHandlers } = require('./server');
+const logger = require('./services/logger');
 
 const app = setupServer();
 
@@ -15,20 +16,19 @@ const app = setupServer();
  * Server setup
  */
 const server = http.createServer(app);
-
 server.listen(process.env.PORT || 3000, () => {
   const address = server.address();
   const host = typeof address === 'string' ? address : address.address;
   const port = typeof address === 'string' ? 0 : address.port;
-  console.info(`Server is running at http://${host}:${port}`);
-  console.info(`Press Ctrl+C to stop the server`);
+  logger.info(`Server is running at http://${host}:${port}`);
+  logger.info(`Press Ctrl+C to stop the server`);
 });
 // setup routes here
 // gracefully handle shutdown
 process.on('SIGINT', () => {
-  console.info('Shutting down server...');
+  logger.info('Shutting down server...');
   server.close(() => {
-    console.info('Server has been shut down gracefully.');
+    logger.info('Server has been shut down gracefully.');
     /* eslint-disable no-process-exit */
     process.exit(0);
   });

--- a/package.json
+++ b/package.json
@@ -26,7 +26,10 @@
     "lint:fix": "eslint . --fix",
     "format": "prettier --check .",
     "format:fix": "prettier --write .",
-    "prepare": "husky"
+    "prepare": "husky",
+    "build": "npm run lint:fix && npm run format:fix",
+    "build:test": "npm run lint && npm run format",
+    "precommit": "npm run build:test"
   },
   "repository": {
     "type": "git",

--- a/server/plugins/crash-handler.js
+++ b/server/plugins/crash-handler.js
@@ -1,12 +1,14 @@
+const logger = require('../../services/logger');
+
 module.exports = () => {
   process.on('uncaughtException', (err) => {
-    console.error('Uncaught Exception:', err);
+    logger.error('Uncaught Exception:', err);
     /* eslint-disable no-process-exit */
     process.exit(1);
   });
 
   process.on('unhandledRejection', (reason) => {
-    console.error('Unhandled Rejection:', reason);
+    logger.error('Unhandled Rejection:', reason);
     /* eslint-disable no-process-exit */
     process.exit(1);
   });

--- a/server/plugins/database.js
+++ b/server/plugins/database.js
@@ -16,16 +16,19 @@ async function connectToMongo() {
     });
 
     isConnected = true;
-    logger.log('✅ Connected to MongoDB via Mongoose');
+    logger.info('✅ Connected to MongoDB via Mongoose');
 
     process.on('SIGINT', async () => {
       await mongoose.connection.close();
-      logger.log('🔌 MongoDB connection closed due to SIGINT');
+      logger.info('🔌 MongoDB connection closed due to SIGINT');
       /* eslint-disable no-process-exit */
       process.exit(0);
     });
   } catch (err) {
-    logger.error('❌ MongoDB connection error:', err);
+    logger.error('❌ MongoDB connection error:', {
+      error: err.message,
+      stack: err.stack,
+    });
     /* eslint-disable no-process-exit */
     process.exit(1);
   }

--- a/server/plugins/database.js
+++ b/server/plugins/database.js
@@ -1,4 +1,5 @@
 const mongoose = require('mongoose');
+const logger = require('../../services/logger');
 
 let isConnected = false;
 
@@ -15,16 +16,16 @@ async function connectToMongo() {
     });
 
     isConnected = true;
-    console.log('✅ Connected to MongoDB via Mongoose');
+    logger.log('✅ Connected to MongoDB via Mongoose');
 
     process.on('SIGINT', async () => {
       await mongoose.connection.close();
-      console.log('🔌 MongoDB connection closed due to SIGINT');
+      logger.log('🔌 MongoDB connection closed due to SIGINT');
       /* eslint-disable no-process-exit */
       process.exit(0);
     });
   } catch (err) {
-    console.error('❌ MongoDB connection error:', err);
+    logger.error('❌ MongoDB connection error:', err);
     /* eslint-disable no-process-exit */
     process.exit(1);
   }

--- a/server/plugins/error-handler.js
+++ b/server/plugins/error-handler.js
@@ -1,6 +1,8 @@
+const logger = require('../../services/logger');
+
 const useErrorHandler = (app) => {
   app.use((err, req, res) => {
-    console.error('Global Error:', err.stack);
+    logger.error('Global Error:', err.stack);
     res.status(err.status || 500).json({
       message: err.message || 'Internal Server Error',
     });

--- a/services/logger.js
+++ b/services/logger.js
@@ -9,9 +9,9 @@ const log = (payload) => {
 };
 
 module.exports = {
-  info: (message) => log({ level: 'info', message }),
-  error: (message) => log({ level: 'error', message }),
-  warn: (message) => log({ level: 'warn', message }),
-  debug: (message) => log({ level: 'debug', message }),
-  trace: (message) => log({ level: 'trace', message }),
+  info: (message, metadata) => log({ level: 'info', message, ...metadata }),
+  error: (message, metadata) => log({ level: 'error', message, ...metadata }),
+  warn: (message, metadata) => log({ level: 'warn', message, ...metadata }),
+  debug: (message, metadata) => log({ level: 'debug', message, ...metadata }),
+  trace: (message, metadata) => log({ level: 'trace', message, ...metadata }),
 };

